snippets/base/models.py

```diff
@@ -660,11 +661,35 @@ class Addon(models.Model):
     modified = models.DateTimeField(auto_now=True)
 
     name = models.CharField(max_length=255, unique=True)
-    url = models.URLField(unique=True)
+    url = models.URLField(
+        unique=True,
+        help_text='Add-on page on https://addons.mozilla.org.'
+    )
     guid = models.CharField(max_length=255, unique=True)
 
     class Meta:
         ordering = ('name', )
 
     def __str__(self):
         return self.name
+
+    @staticmethod
+    def fetch_details(url):
+        if not url.endswith('/'):
+            url += '/'
```

Comment:
I guess it doesn't hurt to re-check for a trailing slash here, but should we be performing this logic both here and in the `clean` method in `forms.py`?